astropy/units/equivalencies.py

```diff
@@ -561,6 +564,344 @@ def mass_energy():
                         ], "mass_energy")
 
 
+@dataclass(frozen=True)
+class GeometricUnitConversionFactors:
```

Comment:
To me it is slightly illogical to define this in terms of string units - I think conceptually it makes more sense to have a `dict` of equivalencies (see below), and here a dict of dict keyed by the units rather than there strings. But I think it can be done more directly by just defining a `dict` of equivalencies -- see main comment.